src/compo/pandora_2ioda.py

```diff
@@ -97,11 +100,44 @@ def read(self):
 
             lats = lats.astype(np.float32)
             lons = lons.astype(np.float32)
-
-            # 2 vertice: surface pressure and top (0)
-            num_vert = 2
-            pressure_vertice = np.zeros([nlocs, num_vert], dtype=np.float32)
-            pressure_vertice[:, 1] = surf_p
+            aq_class = np.zeros(nlocs, dtype=np.int32)
+
+            if self.site_classification:
+                print('site classification file is available')
+                pandora_sites = pd.read_csv(self.site_classification)
+                pandora_lats = pandora_sites['lat']
+                pandora_lons = pandora_sites['lon']
+
+                lat_tol = 0.01
+                lon_tol = 0.01
+                # Filter rows within the tolerance range
+                close_rows = pandora_sites[
+                    (pandora_lats >= lat - lat_tol) & (pandora_lats <= lat + lat_tol)
```

Comment:
I think you could use that instead:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.RegularGridInterpolator.html#:~:text=Interpolate%20and%20extrapolate,%3DNone)

in case we have sharp gradients in the population density this would be a bit cleaner.

Reply:
I don't clearly see how to use interpolation here. Can you give a little more info please?